apps/api/src/domain/models/media-item/commands/publish-media-item.command.integration.spec.ts

```diff
@@ -0,0 +1,193 @@
+
+                    const { id, published } = updatedInstance;
+
+                    expect(id).toBe(unpublishedMediaItem.id);
+
+                    expect(published).toBe(true);
+
+                    expect(
+                        updatedInstance.eventHistory.some(
+                            ({ type }) => type === 'MEDIA_ITEM_CREATED'
+                        )
+                    );
+                },
+            });
+        });
+    });
+
+    describe('when the command payload has an invalid type', () => {
+        describe('when the id is missing', () => {
+            it('should fail', async () => {
+                await assertCommandError(assertionHelperDependencies, {
+                    buildCommandFSA: () => ({
+                        type: commandType,
+                        payload: {},
+                    }),
+                    initialState,
+                    checkError: (error: InternalError) => {
+                        expect(error).toBeInstanceOf(InvalidCommandPayloadTypeError);
+
+                        expect(error.innerErrors[0].toString().includes('id')).toBe(true);
+                    },
+                });
+            });
+        });
+
+        describe('when the id is a number', () => {
+            it('should fail', async () => {
+                await assertCommandError(assertionHelperDependencies, {
+                    buildCommandFSA: () => buildInvalidCommandFSA(767),
+                    initialState,
+                    checkError: (error: InternalError) => {
+                        expect(error).toBeInstanceOf(InvalidCommandPayloadTypeError);
+
+                        expect(error.innerErrors[0].toString().includes('id')).toBe(true);
+                    },
+                });
+            });
+        });
+
+        describe('when the id is an array', () => {
+            it('should fail', async () => {
+                await assertCommandError(assertionHelperDependencies, {
+                    buildCommandFSA: () => buildInvalidCommandFSA(['12', '123']),
+                    initialState,
+                    checkError: (error: InternalError) => {
+                        expect(error).toBeInstanceOf(InvalidCommandPayloadTypeError);
+
+                        expect(error.innerErrors[0].toString().includes('id')).toBe(true);
+                    },
+                });
+            });
+        });
+    });
+
+    describe('when the external state is invalid', () => {
+        describe('when there is no media item with the given id', () => {
+            it('should fail', async () => {
+                await assertCommandError(assertionHelperDependencies, {
+                    buildCommandFSA: () => validCommandFSA,
+                    // We invalidate the command by invalidating the pre-existing state
+                    initialState: buildInMemorySnapshot({}),
+                    checkError: (error: InternalError) => {
```

Comment:
We might want a helper to build the `checkError` logic for a `Resource Not Found` case for all update commands.